lms/djangoapps/grades/api/views.py

```diff
@@ -149,15 +151,26 @@ def get(self, request, course_id):
                 error_code='user_mismatch'
             )
 
-        course = self._get_course(course_id, user, 'load')
+        course = self._get_course(course_id, request.user, 'load')
         if isinstance(course, Response):
             return course
 
+        if request.user.username == username:
+            user = request.user
```

Comment:
My second point was to overwrite the user only when needed - so only in second case.